CMakeLists.txt

```diff
@@ -0,0 +1,272 @@
+cmake_minimum_required(VERSION 3.5)
+project(networkit CXX)
+
+if("${PROJECT_SOURCE_DIR}" STREQUAL "${PROJECT_BINARY_DIR}")
+    message(SEND_ERROR "In-source builds are not allowed.")
+endif("${PROJECT_SOURCE_DIR}" STREQUAL "${PROJECT_BINARY_DIR}")
+
+# BUILD OPTIONS
+option(NETWORKIT_BUILD_TESTS "Build with tests" OFF)
+option(NETWORKIT_LOGGING "Build with logging support" ON)
+option(NETWORKIT_MONOLITH "Build single library (and tests is requested; required for shared lib)" ON)
+option(NETWORKIT_NATIVE "Optimize for native architecture (often better performance)" OFF)
+option(NETWORKIT_WARNINGS "Issue more warnings" OFF)
+option(NETWORKIT_INCLUDESYMLINK "Create symlink to cpp directory" ON)
```

Comment:
Why do we need those symlinks?

Reply:
There is no direct necessity on build such symlinks. They are quite handy when you use the networkit cpp library inside your own program, because you link your libraries as well as the include files to your build directory. (Otherwise you have link the library, but the .h files to the cpp source directory)

This behaviour was copied from the old scons build environment

Reply:
Another quite drastic but much cleaner variant would be to move networkit/cpp to cpp/networkit. This has HUGE advantages:
- No symlinks (by the way, i ported the CPP project to MSVC so this will become relevant)
- If we use Networkit as a subproject in another project by including it as a git submodule, the 3rd-party just includes the header files as #include <networkit/....>
- Includes WITHIN Networkit can be written the same way rather than relative #include "../"

Reply:
I agree that the current directory structure is suboptimal with respect to the header files. However, I do not think that this should be changed in this PR. Let us transition to cmake first before we change the directory structure.

Regarding a MSVC port: Do you plan to make that public? IMO this would be a huge addition to NetworKit. If you want to discuss a MSVC port, we could do that via email.